src/WebUI/Common/KubeSummary.tsx

```diff
@@ -273,11 +273,65 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
 
     private _onSelectionStoreChanged = () => {
         const selectionStoreState = this._selectionStore.getState();
-        this.setState({
-            showSelectedItem: selectionStoreState.showSelectedItem,
-            selectedItem: selectionStoreState.selectedItem,
-            selectedItemType: selectionStoreState.selectedItemType
-        });
+
+        // If the selection only has id and doesn't have the item, we would need to fetch the item
+        if (!!selectionStoreState.itemUID && !selectionStoreState.selectedItem) {
+            let invokedFunc: (() => Promise<V1DaemonSetList | V1ReplicaSetList | V1StatefulSetList>) | undefined = undefined;
+            let item: V1DaemonSet | V1ReplicaSet | V1StatefulSet | undefined = undefined;
+            switch (selectionStoreState.selectedItemType) {
```

Comment:
selectionStoreState.selectedItemType --> i am not liking this modal of checkign for a specific workload, i understand there is no better way, we should see if we can have a better mechanism.